habitat_baselines/rl/ddppo/algo/ddppo.py

```diff
@@ -6,16 +6,36 @@
 
 from typing import Tuple
 
+import numpy as np
 import torch
 from torch import distributed as distrib
 
-from habitat_baselines.common.rollout_storage import RolloutStorage
 from habitat_baselines.rl.ppo import PPO
 
-EPS_PPO = 1e-5
 
+def _recursive_apply(inp, fn):
+    if isinstance(inp, dict):
+        return type(inp)((k, _recursive_apply(v, fn)) for k, v in inp.items())
+    elif isinstance(inp, (tuple, list)):
+        return type(inp)(_recursive_apply(v, fn) for v in inp)
+    else:
+        return fn(inp)
 
-def distributed_mean_and_var(
+
+def _cpu_to_numpy(inp):
+    return _recursive_apply(
+        inp, lambda t: t.numpy() if t.device.type == "cpu" else t
+    )
+
+
+def _numpy_to_cpu(inp):
```

Comment:
I see these as backward compatibility hacks because we are using a very old PyTorch on the CI that hasn't deprecated the single-process multi-device of DDP. I think TensorDict and NDArrayDict (or other possible variants) cover most general cases and I've already gone through the pain to give those proper typing.